client/me/purchases/manage-purchase/index.tsx

```diff
@@ -639,6 +642,76 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! purchase?.isRefundable || ! purchase?.refundText || ! purchase?.mostRecentRenewDate ) {
+			return;
+		}
+
+		const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+		const lastRenewalDate = new Date( purchase.mostRecentRenewDate );
+		const refundPeriodEndDate = new Date(
+			lastRenewalDate.getTime() + purchase.refundPeriodInDays * ONE_DAY_IN_MILLISECONDS
+		);
+
+		const refundText = (() => {
+			// User would receive another free domain credit upon cancelling.
+			if ( isDomainRegistration( purchase ) && ! hasAmountAvailableToRefund( purchase ) ) {
+			  return translate(
+				"If you're unsure about this domain, cancel it before {{refundDate/}} to choose another free domain instead.",
+				{
+					components: {
+						refundDate: <FormattedDate date={ refundPeriodEndDate } format="LL" />,
+					},
+				}
+			);
+			}
+		  
+			// User would receive a financial refund upon cancelling.
+			return translate(
+				'Eligible for a refund of %(refundAmount)s until {{refundDate/}}.',
```

Comment:
Although I like the idea of mentioning the date, I think one reason we may not do that in the current UI (and rather use wording like "within X days of purchasing") is to avoid ambiguity over timezones, time of day, etc.  The situation we would want to avoid is to tell people that they're still eligible for a refund on a particular day, but then at some point before that day ends, the refund option has actually disappeared already.

Reply:
My personal view is that the problem with mentioning the date is that people often still don’t realise what that means in practice (even when we try make it clear!). :) We could always include the time that the refund window expires here to alleviate that concern?

Happy to defer to you though.

Reply:
I wonder if including the exact time might be **too** specific (even if it's an accurate countdown for when the option disappears from Calypso, we wouldn't necessarily always be so strict in practice).

Overall, I feel like it might just be best to stick to the "within X days of purchase or renewal" kind of wording (which is also the wording that was chosen for the documentation at https://wordpress.com/support/refunds/#refund-periods)?  People should be able to figure out when they purchased something -- and if it's not easy for them to figure that out, maybe **that's** the information we should actually be making easier for them to find :smile:

Reply:
I have no strong view on this (not that it would be a decision for me anyway!), but my general assumption is that if we can take a step out of the way for the user, that's even better. So it might be nice to make it easy for the user not to need to calculate the time which the refund period ends - also should lead to less aggrieved users in Support and refunds that way. 

But I also recognise more convenient might not always be better, especially in the case of refunds! 

Let me know if you think I should update it either way.